src/pages/patch/patchRestartModal/PatchStatusCheckboxContainer.tsx

```diff
@@ -13,20 +13,44 @@ export const PatchStatusCheckboxContainer: React.FC<PatchStatusCheckboxContainer
   tasks,
   selectedTasks,
   toggleSelectedTask,
-}) => (
-  <ScrollableContainer data-cy="patch-status-selector-container">
-    {tasks.map((task) => (
-      <TaskStatusCheckbox
-        task={task}
-        selectedTasks={selectedTasks}
-        toggleSelectedTask={toggleSelectedTask}
-        key={`task_checkbox_${task.id}`}
-      />
-    ))}
-  </ScrollableContainer>
-);
+}) => {
+  const possibleListHeight = tasks.length * itemSize;
+  const listHeight =
+    possibleListHeight < maxListHeight ? possibleListHeight : maxListHeight;
+  const toggleHandler = (e) => {
+    // @ts-ignore
+    const { name } = e.target;
+    if (selectedTasks[name] !== undefined) {
+      toggleSelectedTask(name);
+    }
+  };
+  return (
+    // eslint-disable-next-line jsx-a11y/no-static-element-interactions,jsx-a11y/click-events-have-key-events
+    <div data-cy="patch-status-selector-container" onClick={toggleHandler}>
+      <FixedSizeList
+        height={listHeight}
+        itemSize={itemSize}
+        itemCount={tasks.length}
+        itemData={tasks}
+      >
+        {({ style, data, index }) => {
+          const { id: taskId, status, name: displayName } = data[index];
```

Comment:
Could we just pass the task into `<TaskStatusCheckbox />` and let it use the task to render the info it needs?

Reply:
TaskStatusCheckbox is wrapped in `React.memo` which will memoize the render result by shallow comparing the props. Destructuring non-reference values from `data[index]` helps ensure the component only renders once when taskId, status or displayName change as opposed to rendering when the reference to data[index] changes.